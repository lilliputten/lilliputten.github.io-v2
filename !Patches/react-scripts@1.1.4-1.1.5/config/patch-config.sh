#!/sbin/sh
# @overview Patch tool (patch-tool) config file
# @since 2018.02.25, 21:59
# @version 2018.02.25, 21:59

# Root folder
ROOTCD="node_modules"

# Target path relative to root
PATCHCD="react-scripts/config"
