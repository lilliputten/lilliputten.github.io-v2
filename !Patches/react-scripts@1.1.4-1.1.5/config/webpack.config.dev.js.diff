--- ./webpack.config.dev.js.orig	2018-09-10 19:44:07.491077400 +0300
+++ ./webpack.config.dev.js	2018-09-10 19:43:52.138444400 +0300
@@ -80,7 +80,7 @@
     // We placed these paths second because we want `node_modules` to "win"
     // if there are any conflicts. This matches Node resolution mechanism.
     // https://github.com/facebookincubator/create-react-app/issues/253
-    modules: ['node_modules', paths.appNodeModules].concat(
+    modules: [paths.appSrc, 'node_modules', paths.appNodeModules].concat(
       // It is guaranteed to exist because we tweak it in `env.js`
       process.env.NODE_PATH.split(path.delimiter).filter(Boolean)
     ),
