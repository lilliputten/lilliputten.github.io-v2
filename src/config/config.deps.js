/* eslint-disable semi */
({
  mustDeps : [
  ],
  shouldDeps : [
    { elem : 'css' },
    { elem : 'site' },
  ]
})
