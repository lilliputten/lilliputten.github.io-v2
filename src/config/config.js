/* eslint-env commonjs, browser, node */
/* globals modules */
/**
 * @module config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.09.26, 23:11
 * @version 2018.11.29, 01:29
 *
 */

(function(__global){

  /** List of used submodules */
  var __submodules = [
    'css',
    'site',
    'pages',
  ]

  /** makeConfig ** {{{ Make config object from modules list
   * @param {Array} submodules
   * @return {Object}
   */
  function makeConfig (submodules) {

    var config = {}

    submodules.map(function(module, i){
      var id = __submodules[i]
      return config[id] = module
    })

    // Export to global scope
    __global.config = config

    return config
  }/*}}}*/

  // /** modules ** {{{ DEBUG: Fake modules object
  //  */
  // /* eslint-disable no-console, no-debugger */
  // var modules = {
  //   define: function (id, list, cb) {
  //     var provide = function(module) {
  //       console.log('provide', module);
  //     };
  //     console.log('define', id, list, cb);
  //     cb && cb.apply(window, [provide].concat(list));
  //   },
  // };/*}}}*/

  /** Universal export... ** {{{ */
  if (typeof module === 'object' && typeof module.exports === 'object') {

    /** submodules ** {{{ Make config modules list
     */
    var submodules = __submodules.map(function(id){

      // require commonjs module...
      try {
        var data = require('./__' + id + '/config__' + id + '.js')
      }
      catch(err) {
        var msg = 'Cannot load config module \'' + id + '\': '
        console.error(msg, err) // eslint-disable-line no-console
        debugger // eslint-disable-line no-debugger
        throw new Error(msg + (err.message || err))
      }

      // Export indiviudual __submodules...
      module.exports[id] = data

      return data

    })/*}}}*/

    // Make config object & store to global scope
    module.exports = makeConfig(submodules)
    module.exports.__submodules = __submodules

  }/*}}}*/
  /** YM export... ** {{{ */
  if (typeof modules === 'object' && typeof modules.define === 'function') {

    var configModuleNames = __submodules.map(function(id){ return 'config__' + id })

    modules.define('config', configModuleNames, function(provide /*, modules... */) {

      // Make config modules list from passed in arguments
      var submodules = Array.from(arguments).splice(1)

      // Make config object & store to global scope
      provide(makeConfig(submodules))

    })
  }/*}}}*/

})(typeof global !== 'undefined' ? global : typeof module !== 'undefined' ? module : typeof window !== 'undefined' ? window : this)
