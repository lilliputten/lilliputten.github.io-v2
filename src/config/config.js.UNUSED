/* eslint-env commonjs, browser, node */
/**
 * @module config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.09.26, 23:11
 * @version 2018.11.29, 01:29
 *
 */

/** List of used submodules */
var __submodules = [
  'css',
  'site',
  'pages',
]

/** makeConfig ** {{{ Make config object from modules list
 * @param {Array} submodules
 * @return {Object}
 */
function makeConfig (submodules) {

  var config = {}

  submodules.map(function(module, i){
    var id = __submodules[i]
    return config[id] = module
  })

  // Export to global scope
  __global.config = config

  return config
}/*}}}*/

/** submodules ** {{{ Make config modules list
 */
var submodules = __submodules.map(function(id){

  // require commonjs module...
  try {
    var data = require('./__' + id + '/config__' + id + '.js')
  }
  catch(err) {
    var msg = 'Cannot load config module \'' + id + '\': '
    console.error(msg, err) // eslint-disable-line no-console
    debugger // eslint-disable-line no-debugger
    throw new Error(msg + (err.message || err))
  }

  // Export indiviudual __submodules...
  module.exports[id] = data

  return data

})/*}}}*/

// Make config object & store to global scope
module.exports = makeConfig(submodules)
module.exports.__submodules = __submodules
