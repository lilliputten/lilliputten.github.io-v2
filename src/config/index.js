/* eslint-env es6, node, commonjs */

const config = require('./config.js');
// const { css } = config;
//
// export default config;
// export { css };
// export {...config};
module.exports = config;
module.exports.site = config.site;
module.exports.css = config.css;
