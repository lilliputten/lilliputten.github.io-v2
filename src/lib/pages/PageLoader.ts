import PageTools, { IPage, TPageId, TPagePathname, TPageUrl, TPageContent } from 'lib/pages/PageTools';

export { IPage, TPageId, TPagePathname, TPageUrl, TPageContent };

// // TODO:
// // https://blog.logrocket.com/async-rendering-in-react-with-suspense-5d0eaac886c8
// const someFetcher = createResource(async () => {
//   const res = await fetch(`https://api.github.com/search/users?q=yomete`);
//   return await res.json();
// });

export default class PageLoader {

  private pageTools: PageTools;

  /** constructor ** {{{ */
  constructor() {
    // super();
    this.pageTools = new PageTools();
  }/*}}}*/

  /** loadPage ** {{{
   */
  public loadPage(pathname: TPagePathname): Promise<IPage> {

    const url = this.pageTools.normalizeUrl(pathname);
    const id = this.pageTools.normalizeId(url);

    const promise = this.fetchUrl(url)
      .then((content) => {
        return { id, url, content };
      })
      // TODO: Make md parser
      .catch((err) => {
        console.error('PageLoader:loadPage error (promise catch)', err); // tslint:disable-line no-console
        debugger; // tslint:disable-line no-debugger
        return Promise.reject({
          id, url,
          error: 'pageLoadError',
          message: `Cannot load page '${id}'`,
          details: err,
          allowRefetchPage: true,
        });
      })
    ;

    return promise;

  }/*}}}*/

  /** fetchUrl ** {{{
   */
  private fetchUrl(url: string): Promise<any> {
    return new Promise((resolve, reject) => {
      return fetch(url)
        .then((result) => this.pageTools.delayPromise(3000, result))
        .then((res: any) => {
          if (res && res.status !== 200) {
            // tslint:disable-next-line no-console
            console.error('PageLoader:fetchUrl error (invalid response status)', res);
            debugger; // tslint:disable-line no-debugger
            return reject(res);
          }
          res.text()
            .then((content: any) => resolve(content))
          ;
        })
        .catch(reject)
      ;
    });
  }/*}}}*/

}
