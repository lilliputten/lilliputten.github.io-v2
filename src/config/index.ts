/* eslint-env es6, node, commonjs */

// import config from './config.js';
import config = require('./config.js');

// // TODO: Export by `__submodules` list?
// module.exports.default = config;
// if (Array.isArray(config.__submodules)) {
//   config.__submodules.map((id: string) => module.exports[id] = config[id]);
// }
//
// // Static export (due to TS errors; TODO?)
// export default config;
// export const css = config.css;
// export const site = config.site;

// TODO: Make imports compatible with VSCode Intellisense
// export const pages = config.pages;
export const css = require('./__css/config__css');
export const site = require('./__site/config__site');
export const pages = require('./__pages/config__pages');
export default { css, site, pages };
