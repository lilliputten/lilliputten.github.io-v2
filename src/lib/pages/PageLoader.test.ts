
import { pages as pagesConfig } from '../../config';
import PageLoader from './PageLoader';

/** Do test for fail cases? */
const TO_TEST_FAIL_PROMISES = false;

let pageLoader: PageLoader;

describe('PageLoader', () => {

  /*{{{*/beforeAll(() => {
    pageLoader = new PageLoader();
  });/*}}}*/

  /*{{{*/describe('should have basic features', () => {

    /*{{{*/it('should create instance without crashing', () => {
      expect(typeof pageLoader).toBe('object');
    });/*}}}*/

    /*{{{*/it('should have fetchUrl private method', () => {
      expect(typeof (pageLoader as any).fetchUrl).toBe('function');
    });/*}}}*/

    /*{{{*/it('should have loadPage method', () => {
      expect(typeof pageLoader.loadPage).toBe('function');
    });/*}}}*/

  });/*}}}*/

  /*{{{*/describe('loadPage method', () => {

    const successId = '/success';
    const failId = '/fail';

    /*{{{*/beforeAll(() => {
      window.fetch = jest.fn().mockImplementation((url) => {
        const successContent = `
---
frontmatter: value
---

# success

text
`;
        const successResponse = { status: 200, text: () => Promise.resolve(successContent) };
        const failResponse = { status: 500 };
        const res = url.startsWith(pagesConfig.urlPrefix + successId) ? successResponse : failResponse;
        return Promise.resolve(res);
      });
    });/*}}}*/
    /*{{{*/afterAll(() => {
      delete window.fetch;
    });/*}}}*/

    /*{{{*/it('should return promise', () => {
      const successPromise = pageLoader.loadPage(successId);
      expect(successPromise instanceof Promise).toBe(true);
    });/*}}}*/

    /*{{{*/describe('success promise', () => {

      let successPromise: Promise<any>;

      /*{{{*/beforeAll(() => {
        successPromise = pageLoader.loadPage(successId);
      });/*}}}*/

      /*{{{*/it('should resolve object', () => {
        return successPromise
          .then((res) => {
            expect(typeof res).toBe('object');
            expect(res).toMatchSnapshot();
          })
        ;
      });/*}}}*/

    });/*}}}*/

    /*{{{*/TO_TEST_FAIL_PROMISES && describe('fail promise', () => {

      let failPromise: Promise<any>;

      /*{{{*/beforeAll(() => {
        failPromise = pageLoader.loadPage(failId);
      });/*}}}*/

      /*{{{*/it('should reject object', () => {
        return failPromise
          .catch((err) => {
            expect(typeof err).toBe('object');
            // expect(err).toMatchSnapshot();
          })
        ;
      });/*}}}*/

    });/*}}}*/

  });/*}}}*/

});
