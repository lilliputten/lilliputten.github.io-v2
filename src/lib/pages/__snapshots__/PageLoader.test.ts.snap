// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader loadPage method success promise should resolve object 1`] = `
Object {
  "content": <ReactMarkdownWithHtml
    astPlugins={Array []}
    escapeHtml={false}
    plugins={
      Array [
        Array [
          [Function],
        ],
      ]
    }
    rawSourcePos={false}
    renderers={
      Object {
        "code": [Function],
        "customComments": [Function],
        "list": [Function],
        "text": [Function],
      }
    }
    skipHtml={false}
    source="# success

text
"
    sourcePos={false}
    transformLinkUri={[Function]}
  />,
  "frontmatter": Object {
    "frontmatter": "value",
  },
  "id": "/success",
  "source": "
---
frontmatter: value
---

# success

text
",
  "title": "success",
  "url": "/site/success.md",
}
`;
